RealmSwift/Optional.swift

```diff
@@ -43,14 +43,15 @@ extension Bool: RealmOptionalType {}
 
  To change the underlying value stored by a `RealmOptional` instance, mutate the instance's `value` property.
  */
-public final class RealmOptional<Value: RealmOptionalType>: RLMOptionalBase {
+@available(*, deprecated, message: "RealmOptional has been deprecated, use RealmProperty<T?> instead.")
```

Comment:
`@available(*, deprecated, renamed: "RealmProperty")` makes Xcode have a fix-it to automatically update the name.

Reply:
would there be any issues with users having to manually mark the type as optional after? eg RealmOptional<Int> -> RealmProperty<Int?>`

Reply:
I think we can still specify a message with renamed telling them that? It'd be nice if we could do it automatically, but it's probably still worth doing the fixit even if it's incomplete.